lib/src/view_model.dart

```diff
@@ -39,6 +69,7 @@ abstract class ViewModel<State, Effect> {
     }
   }
 
+  /// Emits a new [effect]
```

Comment:
/// Emits a new [Effect] to the ViewModel.
///
/// This method is responsible for handling the emission of effects. It checks
/// if the ViewModel is closed before attempting to emit a new effect.
///
/// Parameters:
///   - [effect]: The new effect to be emitted by the ViewModel.
///
/// If the ViewModel is closed, a debug message is printed, and the effect emission is skipped.
/// Otherwise, the new effect is set, and it is added to the effect controller.
///
/// Throws:
///   - If an error occurs during the effect emission, it is rethrown.